### PR TITLE
convert Yips wiki pages to sphinx docs

### PR DESCRIPTION
**Pull request by jimoibm**
@kadler,
I have the draft done for the conversion for issue #19 .

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+SPHINXPROJ    = XMLSERVICE
+SOURCEDIR     = ./srcdir
```
Comment (kadler) [inline docs/Makefile]: It looks like you re-ran sphinx-quickstart or something (line endings changed to CRLF) and changed the source and build directories. These shouldn't be changed and in fact it doesn't build:

```
$ make html
usage: sphinx-build [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]
sphinx-build: error: config directory doesn't contain a conf.py file (/home/kadler/projects/xmlservice/docs/srcdir)
make: *** [Makefile:20: html] Error 2
```

Please revert the changes to this file and move the rst files in to the docs/ directory.

Comment (jimoibm) [reply docs/Makefile]: I forgot to move conf.py into srcdir. Sorry. Now it works.

Comment (kadler) [reply docs/Makefile]: But why is there a srcdir at all? Sphinx was already set up correctly. Please undo the change and move the rst files in to the docs directory.

Comment (jimoibm) [reply docs/Makefile]: I moved *.rst back and also changed Makefile as you asked. I have new commit submitted.

**Comment by jimoibm** [comment]
@kadler How can I get a account to login the Jekins?

**Comment by kadler** [comment]
I've added you

**Comment by jimoibm** [comment]
@kadler It seems `make -j4` caused the failure. `RUNSQLSTM` started before `CRTSRVPGM` and could not get the lock which resulted in 

> SQL7909  10       1  Position 1 IPLUG4K was created, changed, or dropped, but object not modified.

> 10 level severity errors found in source

I tried on my 7.4 box with `make -j4`. I could get **SQL7909**  but it didn't fail my make right after the sql errors. Do you have any ideas? Can we don't use the parallel make?

**Comment by kadler** [comment]
Aha! The xmlstoredp.sqlinst will result in the generic [sqlinst](https://github.com/IBM/xmlservice/blob/master/Makefile.in#L90) rule running, but it has no dependency on xmlstoredp.srvpgm. Unfortunately, I don't think that we can create another rule that defines the prereq for the srvpgm and have make keep using the generic rule (from what I can tell, make will ignore the generic rule in that case :frowning_face:). Since we only have one SQL script, we could update the existing generic rule to require a matching srvpgm, eg.

```patch
-%.sqlinst: src/%.sql
+%.sqlinst: src/%.sql %.srvpgm
 	system -q "RUNSQLSTM SRCSTMF('$<')" && touch $@
```

**Comment by jimoibm** [comment]
@kadler The dependency you suggested looks working. But now I got this error:

```
> bash-4.4$ make -j4 build-procedure
> (system -q 'CHKOBJ XMLSERVICE *LIB' || system -q 'CRTLIB XMLSERVICE') && touch XMLSERVICE.lib
> system "CRTSQLRPGI OBJ(XMLSERVICE/xmlstoredp) SRCSTMF('src/xmlstoredp.sqlrpgle') OBJTYPE(*MODULE) REPLACE(*YES) RPGPPOPT(*LVL2) COMPILEOPT('INCDIR(''src/'') TGTCCSID(37)')" > xmlstoredp.log 2> xmlstoredp.msg && rm xmlstoredp.log xmlstoredp.msg || touch xmlstoredp.failed
> system "CRTRPGMOD MODULE(XMLSERVICE/plugconf) SRCSTMF('src/plugconf.rpgle') DBGVIEW(*ALL) REPLACE(*YES) TGTCCSID(37)" > plugconf.log 2> plugconf.msg && rm plugconf.log plugconf.msg || touch plugconf.failed
> system "CRTRPGMOD MODULE(XMLSERVICE/plugbug) SRCSTMF('src/plugbug.rpgle') DBGVIEW(*ALL) REPLACE(*YES) TGTCCSID(37)" > plugbug.log 2> plugbug.msg && rm plugbug.log plugbug.msg || touch plugbug.failed
> test ! -e plugconf.msg || cat plugconf.msg
> system "CRTRPGMOD MODULE(XMLSERVICE/plugipc) SRCSTMF('src/plugipc.rpgle') DBGVIEW(*ALL) REPLACE(*YES) TGTCCSID(37)" > plugipc.log 2> plugipc.msg && rm plugipc.log plugipc.msg || touch plugipc.failed
> RNS9313: Library XMLSERVICE cannot be found.
> RNS9309: Compilation failed. Module PLUGCONF not created in library XMLSERVICE.
```

Looks like %.lib was not done and then the other dependencies started executing.  build-main, build-cgi... all have this problem on my box today. Does **system crtlib** command run in a child process and cause make not wait?

**Comment by kadler** [comment]
Looks like all the build rules need to have a dependency on the library just like this one: https://github.com/IBM/xmlservice/blob/master/Makefile.in#L117

**Comment by jimoibm** [comment]
I tried adding dependency of creating library ahead, but it didn't work. I saw CRTLIB was ran ahead, but the following CRTMOD just failed with no library found. So I guessed make didn't wait for CRLIB done. 
Not like svrpgm and sqlint dependency worked.

**Comment by jimoibm** [comment]
The .lib dependency seems not working. I tried your Makefile.in also got this:
```
bash-4.4$ make -j4 build-cgi
(system -q 'CHKOBJ XMLSERVICE *LIB' || system -q 'CRTLIB XMLSERVICE') && touch XMLSERVICE.lib
system "CRTRPGMOD MODULE(XMLSERVICE/xmlcgi) SRCSTMF('src/xmlcgi.rpgle') DBGVIEW(*ALL) REPLACE(*YES) TGTCCSID(37)" > xmlcgi.log 2> xmlcgi.msg && rm xmlcgi.log xmlcgi.msg || touch xmlcgi.failed
system "CRTRPGMOD MODULE(XMLSERVICE/plugconf) SRCSTMF('src/plugconf.rpgle') DBGVIEW(*ALL) REPLACE(*YES) TGTCCSID(37)" > plugconf.log 2> plugconf.msg && rm plugconf.log plugconf.msg || touch plugconf.failed
system "CRTRPGMOD MODULE(XMLSERVICE/plugbug) SRCSTMF('src/plugbug.rpgle') DBGVIEW(*ALL) REPLACE(*YES) TGTCCSID(37)" > plugbug.log 2> plugbug.msg && rm plugbug.log plugbug.msg || touch plugbug.failed
test ! -e xmlcgi.msg || cat xmlcgi.msg
test ! -e plugconf.msg || cat plugconf.msg
RNS9313: Library XMLSERVICE cannot be found.
RNS9309: Compilation failed. Module XMLCGI not created in library XMLSERVICE.
test ! -e xmlcgi.log || ./parse.sh xmlcgi.log
RNS9313: Library XMLSERVICE cannot be found.
```
